services/scheduler/src/main/java/org/mule/service/scheduler/internal/config/ThreadPoolsConfig.java

```diff
@@ -0,0 +1,217 @@
+      logger.info("No " + MULE_HOME_DIRECTORY_PROPERTY + " defined. Using default values for thread pools.");
+      return config;
+    }
+
+    File defaultConfigFile = new File(muleHome, "conf" + separator + "scheduler-pools.conf");
+    if (!defaultConfigFile.exists()) {
+      logger.info("No thread pools config file found. Using default values.");
+      return config;
+    }
+
+    logger.info("Loading thread pools configuration from " + defaultConfigFile.getPath());
+
+    final Properties properties = new Properties();
+    try (final FileInputStream configIs = new FileInputStream(defaultConfigFile)) {
+      properties.load(configIs);
+    } catch (IOException e) {
+      throw new DefaultMuleException(e);
+    }
+
+    ScriptEngineManager manager = new ScriptEngineManager();
+    ScriptEngine engine = manager.getEngineByName("js");
+    engine.put("C", config.cores);
+
+    config.setGracefulShutdownTimeoutSeconds(resolveNumber(properties, PROP_PREFIX + "gracefulShutdownTimeoutSeconds"));
+    config.setCpuLightPoolSize(resolveExpression(properties, PROP_PREFIX + "cpuLight.threadPoolSize", config, engine));
+    config.setIoCorePoolSize(resolveExpression(properties, PROP_PREFIX + "io.coreThreadPoolSize", config, engine));
+    config.setIoMaxPoolSize(resolveExpression(properties, PROP_PREFIX + "io.maxThreadPoolSize", config, engine));
+    config.setIoKeepAliveSeconds(resolveNumber(properties, PROP_PREFIX + "io.threadKeepAliveSeconds"));
+    config.setCpuIntensivePoolSize(resolveExpression(properties, PROP_PREFIX + "cpuIntensive.threadPoolSize", config, engine));
+
+    return config;
+  }
+
+  private static int resolveNumber(Properties properties, String propName) throws DefaultMuleException {
+    final String property = properties.getProperty(propName);
+    try {
+      final int result = parseInt(property);
+      if (result <= 0) {
+        throw new DefaultMuleException(propName + ": Value has to be greater than 0");
+      }
+
+      return result;
+    } catch (NumberFormatException e) {
+      throw new DefaultMuleException(propName + ": " + e.getMessage(), e);
+    }
+  }
+
+  private static int resolveExpression(Properties properties, String propName, ThreadPoolsConfig threadPoolsConfig,
+                                       ScriptEngine engine)
+      throws DefaultMuleException {
+    final String property = properties.getProperty(propName).trim().toUpperCase();
+    if (!POOLSIZE_PATTERN.matcher(property).matches()) {
+      throw new DefaultMuleException(propName + ": Expression not valid");
+    }
+    try {
+      final int result = ((Number) engine.eval(property)).intValue();
+      if (result <= 0) {
+        throw new DefaultMuleException(propName + ": Value has to be greater than 0");
+      }
+
+      return result;
+    } catch (ScriptException e) {
+      throw new DefaultMuleException(propName + ": " + e.getMessage(), e);
+    }
+  }
+
+  private int cores = getRuntime().availableProcessors();
```

Comment:
Any reason this can't be static, although I guess it doesn't make much difference.